File: src/BrewUp.Rest/Modules/SalesModule.cs
using BrewUp.Sales.Facade;
using BrewUp.Sales.Facade.Endpoints;
using Serilog.Core;

namespace BrewUp.Rest.Modules;

public class SalesModule : IModule
{
    public bool IsEnabled => true;
    public int Order => 0;
    
    private Logger _logger = null!;

    public IServiceCollection Register(WebApplicationBuilder builder)
    {
        _logger = Logging.Build(builder.Configuration);
        builder.Services.AddSalesFacade(builder.Configuration);
        
        return builder.Services;
    }

    public WebApplication Configure(WebApplication app)
    {
        // var salesCompositionRoot = SalesCompositionRoot.Build(_logger);
        // app.DefineSalesRoutes(salesCompositionRoot);
        
        return app;
    }
}

File: src/BrewUp.Rest.FSharp/Modules/SalesModule.fs
namespace BrewUp.Rest.FSharp.Modules

open Microsoft.AspNetCore.Builder
open Microsoft.Extensions.DependencyInjection
open Giraffe

type SalesModule() =
    interface IModule with
        member _.IsEnabled = true
        member _.Order = 0
        
        member _.Register(builder: WebApplicationBuilder) =
            BrewUp.Sales.Facade.FSharp.SalesFacadeHelper.addSalesFacade builder.Services builder.Configuration |> ignore
            builder.Services
            
        member _.Configure(app: WebApplication) =
            app